### PR TITLE
ModelExporter: add the possibility to specify and export additional xmlblobs

### LINKED ISSUE #1080: URDFModelExport: Add the possibility add xml blobs
When exporting the urdf of a `iDynTree::Model` it would be nice to add the possibility to include xml blobs

cc @traversaro @mfussi66 @pattacini

### PR DESCRIPTION
**Pull request by Nicogene**
It fixes #1080
Before exporting all [non-printable characters](https://en.cppreference.com/w/cpp/string/byte/isprint) are removed from the blobs, otherwise they mess up the indentation of the whole urdf.

Please review code.

### CODE REVIEW

File: src/model_io/codecs/include/iDynTree/ModelIO/ModelExporter.h [hunk 1]
```diff
@@ -26,9 +26,8 @@ namespace iDynTree
  *
  * Options for the iDynTree exporter.
  */
-class ModelExporterOptions
+struct ModelExporterOptions
 {
-public:
```
Comment (traversaro) [inline src/model_io/codecs/include/iDynTree/ModelIO/ModelExporter.h:31]: What is the rationale of this change? On some platforms this could break code that forward declare this class, but if there is a good reason we can change it.

**Review by traversaro — CHANGES_REQUESTED** [review]
Minor changes.

File: src/model_io/codecs/include/iDynTree/ModelIO/ModelExporter.h (hunk 1 above)
Comment (Nicogene) [reply src/model_io/codecs/include/iDynTree/ModelIO/ModelExporter.h:31]: Since it was a class w/ only public members I thought a struct is more suitable, but I can undo it if it breaks user code!

Comment (traversaro) [reply src/model_io/codecs/include/iDynTree/ModelIO/ModelExporter.h:31]: With a first glance nobody seems to forward declare this at least in public repos, so probaly we can change this without problem.

File: src/model_io/codecs/src/ModelExporter.cpp [hunk 2]
```diff
@@ -21,7 +21,8 @@ namespace iDynTree
 ModelExporterOptions::ModelExporterOptions()
 : baseLink(""),
     exportFirstBaseLinkAdditionalFrameAsFakeURDFBase(true),
-    robotExportedName("iDynTreeURDFModelExportModelName") {}
+    robotExportedName("iDynTreeURDFModelExportModelName"),
+    xmlBlobs{""} {}
```
Comment (traversaro) [inline src/model_io/codecs/src/ModelExporter.cpp:25]: Why we put an empty string in the vector?

Comment (Nicogene) [reply src/model_io/codecs/src/ModelExporter.cpp]: You are right it doesn't have sense, I changed it initializing w/ an empty vector

**Review by traversaro — APPROVED** [review]
